module/zfs/arc.c

```diff
@@ -6900,6 +6900,12 @@ arc_state_multilist_index_func(multilist_t *ml, void *obj)
 	    multilist_get_num_sublists(ml));
 }
 
+#define	WARN_IF_TUN_IGNORED(tun, val) do { \
+	if ((tun) && ((tun) != (val))) \
+		dprintf("WARNING: ignoring tunable %s (using %llu instead)\n",\
+		    (#tun), (val)); \
+} while (0)
```

Comment:
I think it makes sense to include this debugging for development.  But I'd like to drop it for the final version, if we're worried about regressions / bugs in this code I'd suggest we add a ZTS test case instead.

Reply:
I think it is useful for the end-user too. I would love to have a message in the kernel log saying that some of my tuneable are being ignored.

Reply:
The dprintf should be a no-op in non-debug builds:
https://github.com/openzfs/zfs/blob/c96a32e1a30a97212cdfc68e465e55de52a407d5/include/sys/zfs_debug.h#L84-L95

This wouldn't be a good place for messages that are enabled by default IMO, since the function runs periodically.

Reply:
Right.  So this message isn't user visible, nor is it enabled by default even in debug builds.  I could see a case for printing a user visible warning when called by `param_set_arc_long` and `param_set_arc_int` since those are run when the module option is modified.

Reply:
Done. Also printed at module load.